File: quickcontrols2/quickcontrols2.cpp
// +build !minimal

#define protected public
#define private public

#include "quickcontrols2.h"
#include "_cgo_export.h"

#include <QByteArray>
#include <QQuickStyle>
#include <QString>

struct QtQuickControls2_PackedString QQuickStyle_QQuickStyle_Name()
{
	return ({ QByteArray tca3084 = QQuickStyle::name().toUtf8(); QtQuickControls2_PackedString { const_cast<char*>(tca3084.prepend("WHITESPACE").constData()+10), tca3084.size()-10 }; });
}

struct QtQuickControls2_PackedString QQuickStyle_QQuickStyle_Path()
{
	return ({ QByteArray t432960 = QQuickStyle::path().toUtf8(); QtQuickControls2_PackedString { const_cast<char*>(t432960.prepend("WHITESPACE").constData()+10), t432960.size()-10 }; });
}

struct QtQuickControls2_PackedString QQuickStyle_QQuickStyle_AvailableStyles()
{
	return ({ QByteArray td4aa30 = QQuickStyle::availableStyles().join("|").toUtf8(); QtQuickControls2_PackedString { const_cast<char*>(td4aa30.prepend("WHITESPACE").constData()+10), td4aa30.size()-10 }; });
}

void QQuickStyle_QQuickStyle_SetFallbackStyle(struct QtQuickControls2_PackedString style)
{
	QQuickStyle::setFallbackStyle(QString::fromUtf8(style.data, style.len));
}

void QQuickStyle_QQuickStyle_SetStyle(struct QtQuickControls2_PackedString style)
{
	QQuickStyle::setStyle(QString::fromUtf8(style.data, style.len));
}

File: quickcontrols2/quickcontrols2.go
// +build !minimal

package quickcontrols2

import (
	"github.com/bluszcz/cutego"
	"github.com/bluszcz/cutego/core"
	"strings"
	"unsafe"
)

type QQuickStyle struct {
	internal.Internal
}

type QQuickStyle_ITF interface {
	QQuickStyle_PTR() *QQuickStyle
}

func (ptr *QQuickStyle) QQuickStyle_PTR() *QQuickStyle {
	return ptr
}

func (ptr *QQuickStyle) Pointer() unsafe.Pointer {
	if ptr != nil {
		return unsafe.Pointer(ptr.Internal.Pointer())
	}
	return nil
}

func (ptr *QQuickStyle) SetPointer(p unsafe.Pointer) {
	if ptr != nil {
		ptr.Internal.SetPointer(uintptr(p))
	}
}

func PointerFromQQuickStyle(ptr QQuickStyle_ITF) unsafe.Pointer {
	if ptr != nil {
		return ptr.QQuickStyle_PTR().Pointer()
	}
	return nil
}

func (n *QQuickStyle) ClassNameInternalF() string {
	return n.Internal.ClassNameInternalF()
}

func NewQQuickStyleFromPointer(ptr unsafe.Pointer) (n *QQuickStyle) {
	n = new(QQuickStyle)
	n.InitFromInternal(uintptr(ptr), "quickcontrols2.QQuickStyle")
	return
}

func (ptr *QQuickStyle) DestroyQQuickStyle() {
}

func QQuickStyle_AddStylePath(path string) {

	internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_AddStylePath", "", path})
}

func (ptr *QQuickStyle) AddStylePath(path string) {

	internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_AddStylePath", "", path})
}

func QQuickStyle_AvailableStyles() []string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_AvailableStyles", ""}).([]string)
}

func (ptr *QQuickStyle) AvailableStyles() []string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_AvailableStyles", ""}).([]string)
}

func QQuickStyle_Name() string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_Name", ""}).(string)
}

func (ptr *QQuickStyle) Name() string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_Name", ""}).(string)
}

func QQuickStyle_Path() string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_Path", ""}).(string)
}

func (ptr *QQuickStyle) Path() string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_Path", ""}).(string)
}

func QQuickStyle_SetFallbackStyle(style string) {

	internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_SetFallbackStyle", "", style})
}

func (ptr *QQuickStyle) SetFallbackStyle(style string) {

	internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_SetFallbackStyle", "", style})
}

func QQuickStyle_SetStyle(style string) {

	internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_SetStyle", "", style})
}

func (ptr *QQuickStyle) SetStyle(style string) {

	internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_SetStyle", "", style})
}

func QQuickStyle_StylePathList() []string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_StylePathList", ""}).([]string)
}

func (ptr *QQuickStyle) StylePathList() []string {

	return internal.CallLocalFunction([]interface{}{"", "", "quickcontrols2.QQuickStyle_StylePathList", ""}).([]string)
}

func init() {
	internal.ConstructorTable["quickcontrols2.QQuickStyle"] = NewQQuickStyleFromPointer
}
